sdk/storage/azqueue/service/models.go

```diff
@@ -0,0 +1,122 @@
+type RetentionPolicy = generated.RetentionPolicy
+
+// Metrics - a summary of request statistics grouped by API in hour or minute aggregates for queues
+type Metrics = generated.Metrics
+
+// Logging - Azure Analytics Logging settings.
+type Logging = generated.Logging
+
+// TODO: CreateQueueOptions = queue.CreateOptions
+// TODO: DeleteQueueOptions = queue.DeleteOptions
+
+// StorageServiceProperties - Storage Service Properties.
+type StorageServiceProperties = generated.StorageServiceProperties
+
+// StorageServiceStats - Stats for the storage service.
+type StorageServiceStats = generated.StorageServiceStats
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+// ListQueuesOptions provides set of configurations for ListQueues operation
+type ListQueuesOptions struct {
+	Include ListQueuesInclude
+
+	// A string value that identifies the portion of the list of queues to be returned with the next listing operation. The
+	// operation returns the NextMarker value within the response body if the listing operation did not return all queues
+	// remaining to be listed with the current page. The NextMarker value can be used as the value for the marker parameter in
+	// a subsequent call to request the next page of list items. The marker value is opaque to the client.
+	Marker *string
+
+	// Specifies the maximum number of queues to return. If the request does not specify max results, or specifies a value
+	// greater than 5000, the server will return up to 5000 items. Note that if the listing operation crosses a partition boundary,
+	// then the service will return a continuation token for retrieving the remainder of the results. For this reason, it is possible
+	// that the service will return fewer results than specified by max results, or than the default of 5000.
+	MaxResults *int32
+
+	// Filters the results to return only queues whose name begins with the specified prefix.
+	Prefix *string
+}
+
+// ListQueuesInclude indicates what additional information the service should return with each queue.
+type ListQueuesInclude struct {
+	// Tells the service whether to return metadata for each queue.
+	Metadata bool
+}
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+// SetPropertiesOptions provides set of options for Client.SetProperties
+type SetPropertiesOptions struct {
+	// The set of CORS rules.
+	Cors []*CorsRule
+
+	// a summary of request statistics grouped by API in hour or minute aggregates for queues
+	HourMetrics *Metrics
+
+	// Azure Analytics Logging settings.
+	Logging *Logging
+
+	// a summary of request statistics grouped by API in hour or minute aggregates for queues
+	MinuteMetrics *Metrics
+}
+
+func (o *SetPropertiesOptions) format() (generated.StorageServiceProperties, *generated.ServiceClientSetPropertiesOptions) {
```

Comment:
Why are we returning *generated.ServiceClientSetPropertiesOptions if we always return nil? Is this for future use for another method?

Reply:
currently its set to nil, this might change in the future for one reason or another. Rather than directly passing nil to the generated layer just abstract it here and change implementation here in the future if something changes.